### DISCUSSION TITLE
WiFi AP + Async WiFi scanner + Async Web Server running Websockets

### DISCUSSION THREAD

**Discussion by HamzaHajeir** [espressif/arduino-esp32#10105] [Q&A]
Hi, I'm having an issue that did not exist in the previous major release (2.x).

The use case simplified: 
- When we run WiFi AP, and run an async WiFi scanner (like every 10 seconds), the scanner interrupts the served data, and reaches to disconnect the connected station (tried by Windows 11 & an Android device).

I've tested this operation with two versions (2.0.11 and 3.0.3). The 2.0.11 could have some little interrupting of the data transmit which could be unnoticed, and does not disconnect the connected station. 

Couldn't make a reliable MCVE, due to the nature of the bug which require some complex system. But however I can give a PIO environment that can show such an issue, if it would help.

With best,
Hamza Hajeir

**Comment by me-no-dev** [comment]
This does not seem to be something we can control from within the Arduino SDK. Maybe you can poke the guys in the IDF repo?

**Reply by HamzaHajeir** [reply]
Isn't there some API or changes in the way dealing with either AP/STA modes, network interfaces, or WiFi scanning functionalities? I'd need to be confident about that.

**Reply by HamzaHajeir** [reply]
@me-no-dev The answer is further vital in raising an issue in ESP-IDF.
